ui/store/actions.js

```diff
@@ -51,9 +52,37 @@ import * as actionConstants from './actionConstants';
 
 let background = null;
 let promisifiedBackground = null;
+const retryQueue = [];
+
+const submitRequestToBackground = async (method, args) => {
+  if (isManifestV3) {
+    if (background.connectionStream.readable) {
+      await promisifiedBackground[method](...args);
+    } else {
+      retryQueue.push({ method, args });
+    }
+  } else {
+    await promisifiedBackground[method](...args);
+  }
+};
+
+const clearRetryQueue = () => {
```

Comment:
The name `clearRetryQueue` has room for interpretation. My first thought was that this function would erase the retry queue without running the methods. What if we renamed this to something like `runRetryQueue`?